src/WordSearch/WordSearchSolver.ts

```diff
@@ -0,0 +1,98 @@
+import trie = require('trie-prefix-tree');
+import {Point} from './Point';
+import {WordSearchResult} from './WordSearchResult';
+
+export class WordSearchSolver {
+  // The grid of letters which makes up the wordsearch puzzle
+  private _matrix: string[][];
+  // Transpositions to apply when finding words
+  private _directions: number[][];
+  // Potential words to find in the letter grid
+  private _targets: any;  // tslint:disable-line
+
+  constructor(matrix: string[][]) {
+    this._matrix = matrix;
+
+    this._directions = [
+      // Cardinal directions
+      [0, 1], [-1, 0], [1, 0], [0, -1],
+      // Diagonal
+      [1, 1], [-1, -1], [1, -1], [-1, 1]
+    ];
+
+    this._targets = trie([]);
+  }
+
+  findWords(words: string[]): WordSearchResult[] {
+    for (const word of words) {
+      if (word !== '') {
+        this._targets.addWord(word.trim());
+      }
+    }
+    return this.search();
+  }
+
+  private search(): WordSearchResult[] {
+    let results: WordSearchResult[] = [];
+    for (let y = 0; y < this._matrix.length; y++) {
+      for (let x = 0; x < this._matrix[y].length; x++) {
+        const point = new Point(x, y);
+        const pointResults = this.startSearch(point);
+        results = results.concat(pointResults);

```

Comment:
You likely want to push onto the original array rather than creating a new array for each, I think you can use "spread args" to do it, e.g.:

```
results.push(...pointResults);
```

or at worst case:

```
results.push.apply(results, pointResults)
```

Reply:
I don't see anything resolved here?

Reply:
Ideally you want to have `const results: Results[] = []` and not use `concat` at all.

Reply:
Looks like you resolved it below, just want to do the same thing here.

Reply:
Oh, yeah, sorry.  I didn't see that I had done this more than once